controllers/grid/settings/user/form/UserForm.php

```diff
@@ -106,16 +106,28 @@ public function display($request = null, $template = null)
      */
     public function execute(...$functionArgs)
     {
-        if (isset($this->userId)) {
-            Repo::userGroup()->deleteAssignmentsByContextId(Application::get()->getRequest()->getContext()->getId(), $this->userId);
+        parent::execute(...$functionArgs);
+    }
 
-            if ($this->getData('userGroupIds')) {
-                foreach ($this->getData('userGroupIds') as $userGroupId) {
-                    Repo::userGroup()->assignUserToGroup($this->userId, $userGroupId);
-                }
-            }
+    /**
+     * Save the user group assignments
+     */
+    public function saveUserGroupAssignments(): void
+    {
+        if (!isset($this->userId)) {
+            return;
         }
 
-        parent::execute(...$functionArgs);
+        Repo::userGroup()
+            ->deleteAssignmentsByContextId(
+                Application::get()->getRequest()->getContext()->getId(), 
+                $this->userId
+            );
+
+        if ($this->getData('userGroupIds')) {
```

Comment:
This trusts the user-specified `userGroupIds` list; these could be in e.g. another context. (It looks like that was already the case but should be fixed here.)

Reply:
@asmecher bit confused about this requirement . Are you suggesting that we should check it against current context and this group if associated with this context ?

Reply:
I have implemented a context wise check to see if the user group id has any association with the given context, please check .

Reply:
That looks good, thanks!